authentication/typescript/azure/src/azureIdentityAccessTokenProvider.ts

```diff
@@ -0,0 +1,41 @@
+import { GetTokenOptions, TokenCredential } from "@azure/core-auth";
+import { AccessTokenProvider, AllowedHostsValidator, validateProtocol } from '@microsoft/kiota-abstractions';
+
+/** Access token provider that leverages the Azure Identity library to retrieve an access token. */
+export class AzureIdentityAccessTokenProvider implements AccessTokenProvider {
+    /**
+     *@constructor
+     *@param credentials The tokenCredential implementation to use for authentication.
+     *@param scopes The scopes to use for authentication.
+     *@param options The options to use for authentication.
+     *@param allowedHosts The allowed hosts to use for authentication.
+     */
+    public constructor(private readonly credentials: TokenCredential, 
+        private readonly scopes: string[] = ['https://graph.microsoft.com/.default'],
+        allowedHosts: Set<string>= new Set<string>(['graph.microsoft.com', 'graph.microsoft.us', 'dod-graph.microsoft.us', 'graph.microsoft.de', 'microsoftgraph.chinacloudapi.cn', 'canary.graph.microsoft.com']),
```

Comment:
> ``
```suggestion
        allowedHosts?: Set<string>,
```

Reply:
Minor comment:

I would sequence of the parameter list as 

(credentials, scopes, getTokenOptions, allowedHosts). This would work when allowed hosts is an optional factor.

Reply:
Allowed hosts should be optional

Reply:
reordered